templates/documents/single_collection_home.html

```diff
@@ -50,6 +50,19 @@ <h1 class="pg-title">{{ collection.name }} {% if collection.is_index %}<span cla
           {% endif %}
         </div>
       </div>
+
+      <div class="my-2 w-full max-w-md" x-data="fileSearch()">
+        <input
+          x-model="searchQuery"
+          @input.debounce.500ms="triggerSearch"
```

Comment:
This approach is fine, but you can use htmx only to achieve the same. If you dispatch a custom event here instead of calling `triggerSearch`, then you should be able to add that event to the `collection-files-container` and `document_source_{{ document_source.id }}` elements' triggers. You can also use [hx-include](https://htmx.org/attributes/hx-include/) to get the search query.